dashboard/app/models/pd/application/teacher1819_application.rb

```diff
@@ -88,6 +89,17 @@ def send_decision_notification_email
       update!(decision_notification_email_sent_at: Time.zone.now)
     end
 
+    # Updates the associated user's school info with the info from this teacher application
+    # based on these rules in order:
+    # 1. Application has a specific school? always overwrite the user's school info
+    # 2. User doesn't have a specific school? overwrite with the custom school info.
+    def update_user_school_info!
+      if school_id || user.school_info.try(&:school).nil?
+        school_info = get_duplicate_school_info(school_info_attr) || SchoolInfo.create!(school_info_attr)
+        user.update!(school_info: school_info)
```

Comment:
If the user is in an invalid state (and a number of our users are in such a state) then this update will fail for reasons unrelated to school_info I think.

I hate to suggest it but update_column might be justified here

Reply:
Good point. Will change. 

As an aside, I'd really like for us to fix the user data